src/routes/posts.js

```diff
@@ -58,7 +58,89 @@ const fetchPosts = async (req, res, next) => {
     }
 }
 
+// ASSIGN TAG TO AN ARTICLE
+
+const assignTagToPost = async (req, res) => {
+    try {
+        const { pid } = req.params // post id;
+        const { tagId, content, title } = req.body
+
+        const postsTags = await db.query(
+            'INSERT INTO posts_tags ("postId","tagId",content,title) SELECT $1,$2,$3,$4 WHERE NOT EXISTS (SELECT * FROM posts_tags WHERE "postId" = $1 AND "tagId" = $2) RETURNING *',
+            [pid, tagId, content, title]
+        )
+
+        if (!postsTags.rows[0]) {
+            res.status(400).json({
+                message: 'Tag is already assigned to the post',
+            })
+        } else {
+            res.status(202).json({
+                message: 'Tag has been successfully assigned to the post',
+                tag: postsTags.rows[0],
+            })
+        }
+    } catch (error) {
+        console.error(error)
+        res.status(500).json({ message: 'Internal server error', error })
+    }
+}
+
+// GET ALL ARTICLES WITH SAME TAG
+
+const queryPosts = async (req, res) => {
+    try {
+        const { tag } = req.query
+
+        // get all posts' id having the same tag
+        const postsIds = await db.query(
+            'SELECT ("postId") FROM posts_tags WHERE "tagId" = $1',
+            [tag]
+        )
+
+        // query for all feeds same tag
+        const feeds = await Promise.all(
+            postsIds.rows.map(async ({ postId }) => {
+                const post = await db.query(
+                    'SELECT * FROM posts WHERE id = $1',
+                    [postId]
+                )
+                return post.rows[0]
+            })
+        )
+
+        res.status(200).json({
+            status: 'success',
+            data: feeds,
+        })
+    } catch (error) {
+        console.error(error)
+        res.status(500).json({ message: 'Internal server error', error })
+    }
+}
+
+// DELETE TAGS IN AN ARTICLE
+
+const deletePostTags = async (req, res) => {
+    try {
+        const { pid, tid } = req.params
+
+        await db.query(
+            'DELETE FROM posts_tags WHERE "postId" = $1 AND "tagId" = $2',
+            [pid, tid]
+        )
+
+        res.status(200).json({ message: 'Tag has been removed from post' })
+    } catch (error) {
+        console.error(error)
+        res.status(500).json({ message: 'Internal server error', error })
+    }
+}
+
 // Routes
 router.route('/').post(createPost).get(fetchPosts)
+router.route('/query').get(queryPosts)
+router.route('/:pid/tags').post(assignTagToPost)
```

Comment:
i think as a parameter, using `:postId` becomes readable in itself to people who come in later as well so we will not even have to comment the code to say it means post id like we've done on line https://github.com/tbf-challenge/teamwork-be/blob/b4b7b11512c0ec08349c70219da7525e55d134d3/src/routes/posts.js#L65

Reply:
Resolved
I have renamed to follow the 'postId' pattern